### PR TITLE
Monitor, publish, record and display web login events

### PR DESCRIPTION
**Pull request by rgooch**

### CODE REVIEW

File: cmd/keymasterd/main.go [hunk 1]
```diff
@@ -1973,6 +1978,7 @@ func (state *RuntimeState) u2fTokenManagerHandler(w http.ResponseWriter, r *http
 	returnAcceptType := getPreferredAcceptType(r)
 	switch returnAcceptType {
 	case "text/html":
+		eventNotifier.PublishWebLoginEvent(authUser)
```
Comment (cviecco) [inline cmd/keymasterd/main.go]: There is no weblogin here.. remove this

File: cmd/keymasterd/main.go [hunk 2]
```diff
@@ -1706,10 +1710,11 @@ func (state *RuntimeState) u2fSignResponse(w http.ResponseWriter, r *http.Reques
 
 			_, err = state.updateAuthCookieAuthlevel(w, r, currentAuthLevel|AuthTypeU2F)
 			if err != nil {
-				logger.Printf("Autch Cookie NOT found ? %s", err)
+				logger.Printf("Auth Cookie NOT found ? %s", err)
 				state.writeFailureResponse(w, r, http.StatusInternalServerError, "Failure updating vip token")
 				return
 			}
+			eventNotifier.PublishAuthEvent(eventmon.AuthTypeU2F, authUser)
```
Comment (cviecco) [inline cmd/keymasterd/main.go]: For consistency on VIPAuthHandler and u2fSignResponse publish the auth event before or after updating the cookie.  (right now one is before and the other after)

File: cmd/keymasterd/main.go [hunk 3]
```diff
@@ -1706,10 +1710,11 @@ func (state *RuntimeState) u2fSignResponse(w http.ResponseWriter, r *http.Reques
 
 			_, err = state.updateAuthCookieAuthlevel(w, r, currentAuthLevel|AuthTypeU2F)
 			if err != nil {
-				logger.Printf("Autch Cookie NOT found ? %s", err)
+				logger.Printf("Auth Cookie NOT found ? %s", err)
```
Comment (cviecco) [inline cmd/keymasterd/main.go]: thanks for the typo fix

File: cmd/keymasterd/main.go (hunk 2 above)
Comment (rgooch) [reply cmd/keymasterd/main.go]: OK.

File: cmd/keymasterd/main.go (hunk 1 above)
Comment (rgooch) [reply cmd/keymasterd/main.go]: Where else am I supposed to get a web login event when U2F is required?

**Pull request merged**